src/EditorFeatures/Core/Implementation/CommentSelection/CommentUncommentSelectionCommandHandler.cs

```diff
@@ -171,51 +194,84 @@ internal void CollectEdits(ICommentUncommentService service, NormalizedSnapshotS
         /// <summary>
         /// Add the necessary edits to comment out a single span.
         /// </summary>
-        private void CommentSpan(ICommentUncommentService service, SnapshotSpan span, List<TextChange> textChanges, List<ITrackingSpan> trackingSpans)
+        private void CommentSpan(ICommentSelectionService service, SnapshotSpan span, List<TextChange> textChanges, List<ITrackingSpan> trackingSpans)
         {
-            var firstAndLastLine = DetermineFirstAndLastLine(span);
+            var (firstLine, lastLine) = DetermineFirstAndLastLine(span);
 
-            if (span.IsEmpty && firstAndLastLine.Item1.IsEmptyOrWhitespace())
+            if (span.IsEmpty && firstLine.IsEmptyOrWhitespace())
             {
+                // No selection, and on an empty line, don't do anything.
                 return;
             }
 
             if (!span.IsEmpty && string.IsNullOrWhiteSpace(span.GetText()))
             {
+                // Just whitespace selected, don't do anything.
                 return;
             }
 
-            if (span.IsEmpty || string.IsNullOrWhiteSpace(span.GetText()))
+            // Get the information from the language as to how they'd like to comment this region.
+            var commentInfo = service.GetInfo(span.Snapshot.AsText(), span.Span.ToTextSpan());
+            if (!commentInfo.SupportsBlockComment && !commentInfo.SupportsSingleLineComment)
+            {
+                // Neither type of comment supported.
+                return;
+            }
+
+            if (commentInfo.SupportsBlockComment && !commentInfo.SupportsSingleLineComment)
+            {
+                // Only block comments supported here.  If there is a span, just surround that
+                // span with a block comment.  If tehre is no span then surround the entire line 
+                // with a block comment.
+                if (span.IsEmpty)
```

Comment:
Ryan, this is to handle your scenario of the user being at the beginning of a line in jsx code.  If they invoke the 'comment selection' command, and they haven't actually selected something, then we will actually surround that line with <!-- -->.  I figured that was the behavior you wanted.  Otherwise users would have to manually make a selection even if they didn't want to.

Note: if the user wants to comment out something that is multiple lines long, they will have to actually select everything.  There is no facility for the language to say "here is the span i want block commented if there is no selection".  If you think that is important, then let me know and we can come up with a suitable way to support that on the interface.